rust/src/pgsql/pgsql.rs

```diff
@@ -341,16 +358,30 @@ impl PgsqlState {
             );
             match PgsqlState::state_based_req_parsing(self.state_progress, start) {
                 Ok((rem, request)) => {
-                    sc_app_layer_parser_trigger_raw_stream_reassembly(flow, Direction::ToServer as i32);
                     start = rem;
-                    if let Some(state) = PgsqlState::request_next_state(&request) {
+                    let (temp_state, tx_completed) = if let Some(state) = PgsqlState::request_next_state(&request) {
```

Comment:
what is temp about the state?

Could it just be `let (state, tx_completed) = if let Some(state) = ....`?

Reply:
Pressing https://github.com/OISF/suricata/pull/11635/files/057c398f669c5b174d555116c87c8d1689db98cf#r1734188284
```rust
let new_state = PgsqlState::request_next_state(&request);
if let Some(state) = new_state {
    self.state_progress = state;
}
...
if let Some(tx) = self.find_or_create_tx() {
    tx.request = Some(request);
    if let Some(state) = new_state {
        if  Self::request_is_complete(state) && ...
```